src/mctpd.c

```diff
@@ -2734,19 +2818,150 @@ static int bus_endpoint_get_prop(sd_bus *bus,
 	return rc;
 }
 
+static int bus_link_get_prop(sd_bus *bus,
+		const char *path, const char *interface, const char *property,
+		sd_bus_message *reply, void *userdata, sd_bus_error *berr)
+{
+	ctx *ctx = userdata;
+	char *tmpstr = NULL;
+	char *link_name = NULL;
+	linkmap_entry *lm = NULL;
+	int rc = 0;
+
+	if (!is_interfaces_path(path)) {
+		sd_bus_error_setf(berr, SD_BUS_ERROR_INVALID_ARGS,
+				"Invalid Object Path");
+		goto out;
+	}
+
+	rc = sd_bus_path_decode_many(path, MCTP_DBUS_PATH "/%/%", &tmpstr,
+		&link_name);
+	if (rc <= 0) {
+		sd_bus_error_setf(berr, SD_BUS_ERROR_INVALID_ARGS,
+				"Invalid Object Path");
+		goto out;
+	}
+
+	lm = find_link_by_name(ctx, link_name);
+	if (!lm) {
+		rc = -ENOENT;
+		goto out;
+	}
+	dfree(tmpstr);
+	dfree(link_name);
+
+	if (lm->published && strcmp(property, "Role") == 0) {
+		rc = sd_bus_message_append(reply, "s", role_name[lm->role]);
+	} else {
+		sd_bus_error_setf(berr, SD_BUS_ERROR_INVALID_ARGS,
+				"Unknown property.");
+		rc = -ENOENT;
+	}
+
+out:
+	set_berr(ctx, rc, berr);
+	return rc;
+}
+
+static int bus_link_set_prop(sd_bus *bus,
+		const char *path, const char *interface, const char *property,
+		sd_bus_message *value, void *userdata, sd_bus_error *berr)
+{
+	ctx *ctx = userdata;
+	const char *state;
+	char *tmpstr = NULL;
+	char *link_name = NULL;
+	linkmap_entry *lm;
+	bool found = false;
+	size_t i;
+	int rc;
+
+	if (!is_interfaces_path(path)) {
+		sd_bus_error_setf(berr, SD_BUS_ERROR_INVALID_ARGS,
+				"Invalid Object Path");
+		goto out;
+	}
+
+	rc = sd_bus_path_decode_many(path, MCTP_DBUS_PATH "/%/%", &tmpstr,
+		&link_name);
+	if (rc <= 0) {
+		sd_bus_error_setf(berr, SD_BUS_ERROR_INVALID_ARGS,
+				"Invalid Object Path");
+		goto out;
+	}
+
+	lm = find_link_by_name(ctx, link_name);
+	if (!lm) {
+		rc = -ENOENT;
+		goto out;
+	}
+
+	dfree(tmpstr);
+	dfree(link_name);
+
+
+	if (strcmp(property, "Role") == 0) {
+		if (lm->role != MCTP_ENTITY_ROLE_UNKNOWN)
+		{
+			sd_bus_error_setf(berr, SD_BUS_ERROR_INVALID_ARGS,
+						"Role is already set.");
+			rc = -ENOENT;
+			goto out;
+		}
```

Comment:
No I don't change  indentation styles. Still use tab.

Reply:
you're flipping between:

```c
    if (thing) {
        stuff;
    } else {
        other_stuff;
    }
```

and

```c
    if (thing)
    {
        stuff;
    }
    else
    {
        other_stuff;
    }
```

I'd prefer the former if possible.

(there *may* be some inconsistency in the code already, but I'd like to avoid adding more)

Reply:
> you're flipping between:
> 
> ```c
>     if (thing) {
>         stuff;
>     } else {
>         other_stuff;
>     }
> ```
> 
I updated code to use this style which is mostly used by the mctpd code.